### PR TITLE
Add CBMC proof of ingestDataBlock

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of ingestDataBlock

Description
-----------
This PR adds files required for the CBMC proof of ingestDataBlock function

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is formatted using Uncrustify.
- [ ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "files.associations": {
+        "cstdlib": "cpp"
+    }
+}
```
Comment (gshvang) [inline .vscode/settings.json:5]: New line missing

Comment (Bujain) [reply .vscode/settings.json]: This file is not required and it is removed it in fb2a3eb

**Review by gshvang — DISMISSED** [review]

**Review by cobusve — DISMISSED** [review]

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                                       const uint8_t * pRawMsg,
+                                       uint32_t messageSize,
+                                       OtaPalStatus_t * pCloseResult );
+
+IngestResult_t ingestDataBlockCleanup( OtaFileContext_t * pFileContext,
+                                       OtaPalStatus_t * pCloseResult )
+{
+    IngestResult_t result;
+
+    __CPROVER_assume( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ) );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:43]: Why is this assumption on the result required? I think `ingestDataBlock` can handle all possible values. Do correct me if I am wrong.

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+IngestResult_t decodeAndStoreDataBlock( OtaFileContext_t * pFileContext,
+                                        const uint8_t * pRawMsg,
+                                        uint32_t messageSize,
+                                        uint8_t ** pPayload,
+                                        uint32_t * pBlockSize,
+                                        uint32_t * pBlockIndex )
+{
+    IngestResult_t result;
+
+    __CPROVER_assume( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ) );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:63]: Similarly here: Why is this assumption on the result required? I think `ingestDataBlock` can handle all possible values. Do correct me if I am wrong.

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+IngestResult_t processDataBlock( OtaFileContext_t * pFileContext,
+                                 uint32_t uBlockIndex,
+                                 uint32_t uBlockSize,
+                                 OtaPalStatus_t * pCloseResult,
+                                 uint8_t * pPayload )
+{
+    IngestResult_t result;
+
+    __CPROVER_assume( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ) );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:93]: And here.

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    __CPROVER_assume( messageSize <= OTA_DATA_BLOCK_SIZE );
+
+    /* CBMC preconditions. */
+
+    /* Havoc otaAgent to non-deterministically set all the bytes in
+     * the structure. */
+    __CPROVER_havoc_object( &otaAgent );
+
+    otaAgent.fileContext.decodeMemMaxSize = decodeMemMaxSize;
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:124]: Why is this required if we are havoc-ing the object?

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    otaInterface.os.mem.free = freeMemStub;
+    otaAgent.pOtaInterface = &otaInterface;
+
+    result = ingestDataBlock( &fileContext, rawMsg, messageSize, &closeResult );
+
+    __CPROVER_assert( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ),
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:131]: Since the function can return any value returned by the functions it calls, we can do without this assert IMO. Correct me if I am wrong.

File: test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                      "Error: Expected a non-NUll value for pRawMsg" );
+    __CPROVER_assert( pPayload != NULL,
+                      "Error: Expected a non-NUll value for pPayload" );
+    __CPROVER_assert( pBlockSize != NULL,
+                      "Error: Expected a non-NUll value for pBlockSize" );
+    __CPROVER_assert( pBlockIndex != NULL,
+                      "Error: Expected a non-NUll value for pBlockIndex" );
+
+    *pPayload = ( uint8_t * ) malloc( 1UL << otaconfigLOG2_FILE_BLOCK_SIZE );
+    __CPROVER_assume( *pPayload != NULL );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c:80]: why should this be non-NULL?
I think that there is a guard against that.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]
